source/Calamari.Common/Features/StructuredVariables/StructuredConfigVariablesService.cs

```diff
@@ -78,26 +83,99 @@ List<string> MatchingFiles(RunningDeployment deployment, string target)
             return files;
         }
 
-        IFileFormatVariableReplacer GetReplacerForFile(string filePath, bool supportNonJsonReplacement)
+        IFileFormatVariableReplacer[] GetReplacersToTryForFile(string filePath, string? specifiedFileFormat, bool supportNonJsonReplacement)
         {
             if (!supportNonJsonReplacement)
             {
-                return jsonReplacer;
+                return new []
+                {
+                    jsonReplacer
+                };
             }
 
-            log.Info($"Feature toggle flag {ActionVariables.StructuredConfigurationFeatureFlag} detected. Trying replacers for all supported file formats.");
+            log.Info($"Feature toggle flag {ActionVariables.StructuredConfigurationFeatureFlag} detected. Considering replacers for all supported file formats.");
+
+            if (!string.IsNullOrWhiteSpace(specifiedFileFormat))
+            {
+                var specifiedReplacer = FindReplacerForFormat(specifiedFileFormat);
+
+                return new []
+                {
+                    specifiedReplacer
+                };
+            }
             
-            // TODO: when we support explicit specification of file formats, handle that here.
+            var guessBasedOnFilePath = FindBestReplacerForFilePath(filePath);
+            if (guessBasedOnFilePath != null)
+            {
+                return new []
+                {
+                    // For backwards compatibility, always try JSON first.
+                    jsonReplacer,
+                    guessBasedOnFilePath
+                };
+            }
 
-            var replacer = replacers.FirstOrDefault(r => r.IsBestReplacerForFileName(filePath));
-            if (replacer != null)
+            return new []
             {
-                log.Info($"The config file at '{filePath}' is being handled as format '{replacer.FileFormatName}' because of the filename.");
-                return replacer;
+                jsonReplacer
+            };
+        }
+
+        IFileFormatVariableReplacer FindReplacerForFormat(string specifiedFileFormat)
```

Comment:
If we changed this to `TryFindReplacerForFormat` would that make it clearer from reading that it can throw an exception?